### ISSUE TITLE
Audit of CSS class change / Update VerticalNavigation markup

### ISSUE THREAD

**Issue by interactivellama** [salesforce/design-system-react#1569]
**TL:DR; With the update to modified-BEM CSS class names, based upon SLDS CSS rule changes, vertical navigation and datepicker were the only component found to have a bad css name converting `--` to `_`.** I will file a bug with SLDS.

https://releasenotes.docs.salesforce.com/en-us/summer17/release-notes/rn_lds.htm

Image captures of 222 stories of components in Storybook between the pre-CSS change found in https://github.com/salesforce/design-system-react/tree/0.8.x and the post-CSS change found in master branch show that `VerticalNavigation` is the only one with a class name discrepancy.

The following were not tested due to DOM use and will be tested manually.
```
// export AppLauncher from '../components/app-launcher/__docs__/storybook-stories';
// export GlobalNavigationBar from '../components/global-navigation-bar/__docs__/storybook-stories';
// export DatePicker from '../components/date-picker/__docs__/storybook-stories';
// export Modal from '../components/modal/__docs__/storybook-stories';
// export Panel from '../components/panel/__docs__/storybook-stories';
// export Popover from '../components/popover/__docs__/storybook-stories';
// export Spinner from '../components/spinner/__docs__/storybook-stories';
// export TimePicker from '../components/time-picker/__docs__/storybook-stories';
```
Snapshot diff file
[__image_snapshots__.zip](https://github.com/salesforce/design-system-react/files/2406721/__image_snapshots__.zip)

These components above where manually tested none had visible issues related to CSS class change.

![screen shot 2018-09-21 at 11 36 32 am](https://user-images.githubusercontent.com/1290832/45907956-738bb280-bdb7-11e8-825d-fbed748065f7.png)
